exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchExporterFaultToleranceIT.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.exporter;
+
+import io.zeebe.exporter.record.Record;
+import io.zeebe.exporter.record.value.MessageRecordValue;
+import io.zeebe.protocol.intent.MessageIntent;
+import io.zeebe.test.util.TestUtil;
+import io.zeebe.test.util.record.RecordingExporter;
+import org.junit.Test;
+
+public class ElasticsearchExporterFaultToleranceIT
+    extends AbstractElasticsearchExporterIntegrationTest {
+
+  @Test
+  public void shouldExportEvenIfElasticNotInitiallyReachable() {
+    // given
+    configuration = getDefaultConfiguration();
+    configuration.index.prefix = "zeebe";
+
+    // when
+    exporterBrokerRule.configure("es", ElasticsearchExporter.class, configuration);
+    exporterBrokerRule.start();
+    exporterBrokerRule.publishMessage("message", "123");
+    elastic.start();
+
+    // then
+    esClient = createElasticsearchClient(configuration);
+    final Record<MessageRecordValue> messageRecord =
+        RecordingExporter.messageRecords(MessageIntent.PUBLISHED)
```

Comment:
Do we need to test that also Publish was seen by the exporter?

Reply:
We test below that this was exported to elastic in the `TestUtil.waitUntil`

Reply:
This waits on Published right? But you can't be sure whether it has seen the Publish command or not?

Reply:
But might be tested somewhere else

Reply:
I see what you mean. Yes, it only waits on published, and assumes that publish was seen. And yes, it's tested elsewhere, here it's more that we ensure that we eventually export after elastic appears, but I will add it anyway for completeness.

Reply:
Thanks :+1: